src/sssom/context.py

```diff
@@ -25,108 +22,79 @@
-    """
-    converter = Converter.from_extended_prefix_map(EXTENDED_PREFIX_MAP)
-    return {record.prefix: record.uri_prefix for record in converter.records}
+def get_internal_converter() -> Converter:
+    """Get a converter from the SSSOM internal context."""
+    context = _get_jsonld_context()
+    prefix_map = {}
+    for key in context["@context"]:
+        v = context["@context"][key]
+        if isinstance(v, str):
+            prefix_map[key] = v
+        elif isinstance(v, dict):
+            if "@id" in v and "@prefix" in v:
+                if v["@prefix"]:
+                    prefix_map[key] = v["@id"]
+    del prefix_map["@vocab"]
+    return Converter.from_prefix_map(prefix_map)
 
 
-def get_built_in_prefix_map() -> PrefixMap:
-    """Get built-in prefix map from the sssom_context variable in the auto-generated 'internal_context.py' file.
+def get_external_converter() -> Converter:
+    """Get prefix map from bioregistry (obo.epm.json)."""
+    return Converter.from_extended_prefix_map(EXTENDED_PREFIX_MAP)
 
-    [Auto generated from sssom.yaml by jsonldcontextgen.py]
 
-    :return: Prefix map
-    """
-    contxt = get_jsonld_context()
+def get_built_in_converter() -> Converter:
+    """Get built-in prefix map from the sssom_context variable in the auto-generated 'internal_context.py' file."""
+    context = _get_jsonld_context()
     prefix_map = {}
-    for key in contxt["@context"]:
+    for key in context["@context"]:
         if key in list(SSSOM_BUILT_IN_PREFIXES):
-            v = contxt["@context"][key]
+            v = context["@context"][key]
             if isinstance(v, str):
                 prefix_map[key] = v
-    return prefix_map
+    return Converter.from_prefix_map(prefix_map)
 
 
 def add_built_in_prefixes_to_prefix_map(
-    prefix_map: Optional[PrefixMap] = None,
-) -> PrefixMap:
+    prefix_map: Union[Converter, PrefixMap, None] = None,
+) -> Converter:
     """Add built-in prefix map from the sssom_context variable in the auto-generated 'internal_context.py' file.
 
-    [Auto generated from sssom.yaml by jsonldcontextgen.py]
-
     :param prefix_map: A custom prefix map
     :raises ValueError: If there is a prefix map mismatch.
     :return: A prefix map
     """
-    builtinmap = get_built_in_prefix_map()
-    if not prefix_map:
-        prefix_map = builtinmap
+    if prefix_map is None:
+        return get_built_in_converter()
+    if isinstance(prefix_map, Converter):
+        converter = prefix_map
     else:
-        for k, v in builtinmap.items():
-            if k not in prefix_map:
-                prefix_map[k] = v
-            elif builtinmap[k] != prefix_map[k]:
-                raise ValueError(
-                    f"Built-in prefix {k} is specified ({prefix_map[k]}) but differs from default ({builtinmap[k]})"
-                )
-    return prefix_map
+        converter = Converter.from_prefix_map(prefix_map)
+    return curies.chain([converter, get_built_in_converter()])
```

Comment:
cool this is possible but not needed anymore: we just use the bioregistry converter and make sure through unit testing that it never violates our prefix preferences when updating the EPM.